migrations/kubernetes-agent/KubernetesAgent.UpgradeManager/KubernetesAgentUpgradeManager.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System.Dynamic;
+using System.Text;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using Octopus.Versioning;
+
+namespace KubernetesAgent.UpgradeManager;
```

Comment:
Everything that's in this file is the public interface of this library. Everything else is/should be internal.